examples/specs_integration/src/sync/entity_delete.rs

```diff
@@ -0,0 +1,37 @@
+use crate::sync::GdSpatial;
+use godot::*;
+use helper::proc_add_remove_events;
+use specs::prelude::*;
+use specs::world::Index;
+use std::collections::HashMap;
+
+pub struct GDDeleteOldSpatials {
+    gdspatial_reader: ReaderId<ComponentEvent>,
+}
+impl GDDeleteOldSpatials {
+    pub fn new(world: &mut specs::World) -> GDDeleteOldSpatials {
+        GDDeleteOldSpatials {
+            gdspatial_reader: world.write_storage::<GdSpatial>().register_reader(),
+        }
+    }
+
+    pub fn apply(
+        &mut self,
+        gd_spatials: ReadStorage<GdSpatial>,
+        spatials: &mut HashMap<Index, Spatial>,
+    ) {
+        let mut new_gd_spatials = BitSet::new();
+        let mut removed_gd_spatials = BitSet::new();
+
+        let events = gd_spatials.channel().read(&mut self.gdspatial_reader);
+        proc_add_remove_events(&mut new_gd_spatials, &mut removed_gd_spatials, events);

```

Comment:
The `proc_add_remove_events` function is only used here (in the example), so it might be good to move the function into this module for ease of navigation.